### PR TITLE
IGNITE-23582 Fix missing memory and GC settings

### PR DESCRIPTION
**Pull request by ivanzlenko**
https://issues.apache.org/jira/browse/IGNITE-23582

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: packaging/common/linux/bootstrap-functions.sh [hunk 1]
```diff
@@ -34,9 +34,15 @@ export LOGGING_JAVA_OPTS="
 
 export CLASSPATH="-classpath @INSTALL_DIR@/lib/@APP_JAR@:@INSTALL_DIR@/lib/* @MAIN_CLASS@"
 
+export JAVA_MEMORY_OPTIONS="-Xmx${JVM_MAX_MEM} -Xms${JVM_MIN_MEM}"
```
Comment (ibessonov) [inline packaging/common/linux/bootstrap-functions.sh:37]: Nice, now please fix it in `bootstrap-functions.bat` as well

**Review by ibessonov — APPROVED** [review]
I'm good with it

**Review by ibessonov — APPROVED** [review]

File: packaging/docker/Dockerfile [hunk 2]
```diff
@@ -15,6 +15,13 @@ ENV JAVA_HOME=/opt/java/openjdk
 ENV PATH="${JAVA_HOME}/bin:${PATH}"
 COPY --from=jre-build /javaruntime $JAVA_HOME
 
+# Set default memory and GC configuration
+ENV JVM_MAX_MEM="16384m"
```
Comment (rpuch) [inline packaging/docker/Dockerfile:19]: How about `16g` instead of `16384m`?

File: packaging/docker/docker-compose.yml [hunk 3]
```diff
@@ -20,6 +20,9 @@ name: ignite3
 x-ignite-def:
   &ignite-def
   image: apacheignite/ignite3:${IGNITE3_VERSION:-latest}
+  environment:
+    JVM_MAX_MEM: "4096m"
```
Comment (rpuch) [inline packaging/docker/docker-compose.yml:24]: 4g?

**Review by rpuch — APPROVED** [review]

**Review by rpuch — APPROVED** [review]

**Pull request merged**